### PR TITLE
Roles

### PR DESCRIPTION
**Pull request by tingamapuro04**
- Install CanCanCan in your project.
- Add a role column to the users table. Remember to use a migration for this.
- A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
 - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
 - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

**Review by Davidosky007 — COMMENTED** [review]
Hi @tingamapuro04 @MoRaad97 👋, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that doesn't show contribution from your partner as this is a paired programming activity, kindly work with your partner on this project 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: db/migrate/20221013090105_remove_role_from_users.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class RemoveRoleFromUsers < ActiveRecord::Migration[7.0]
+  def change
+    remove_column :users, :role, :string
+  end
+end
```
Comment (Davidosky007) [inline db/migrate/20221013090105_remove_role_from_users.rb]: * According to the [project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_add_authorization.md#:~:text=Add%20a%20role%20column%20to%20the%20users%20table.%20Remember%20to%20use%20a%20migration%20for%20this.) you are to  Add a role column to the user's table. Remember to use a migration for this. but here you are removing the column kindly fix this 🙏

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -26,7 +26,7 @@ def create
       format.html do
         if @post.save
           flash[:success] = 'Success'
-          redirect_to user_posts_path(current_user, @post)
+          redirect_to user_posts_path(current_admin, @post)
         else
```
Comment (Davidosky007) [inline app/controllers/posts_controller.rb]: * When I signup successfully I am not redirected to the root route even after adding some users through the rails console below is a recording of the bug I faced 
image

Uploading scrnli_10_13_2022_5-54-09 PM.mp4…

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
## Highlights
* No linter errors 👍 
* Well written PR description 👍 
* Proper use of Gitflow 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidsoky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tingamapuro04** [comment]
I have implemented the requested changes by the reviewer. Any other suggestions are highly welcomed.

File: db/migrate/20221013090105_remove_role_from_users.rb (hunk 1 above)
Comment (codepantha) [inline db/migrate/20221013090105_remove_role_from_users.rb]: - [ ] An error occurs in your migrations as a result of this. Are you sure you want to remove the `role` column? Please see the error below and try to fix it 👇 
image

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @tingamapuro04 👋, 

Great work on making the changes requested by the previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

## Highlights 🎯 
✔️ Descriptive PR
✔️ Used GitFlow correctly
✔️ No linter errors

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by tingamapuro04** [comment]
I have rollback the previous migration and now there's no error

**Review by KDlamini — APPROVED** [review]
### Hi @tingamapuro04 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Good PR description
✔️ Authorization rules implemented
✔️ Correct use of Gitflow
✔️ No linter errors

# Status: Approved ✔️ 

simon

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**